aten/src/ATen/core/ivalue_inl.h

```diff
@@ -868,12 +869,12 @@ TORCH_API intrusive_ptr<ivalue::Future> collectAny(
 // User-defined object.
 struct C10_EXPORT ivalue::Object final : c10::intrusive_ptr_target {
  public:
-  Object(StrongTypePtr type, size_t numSlots) : type_(std::move(type)) {
+  Object(WeakOrStrongTypePtr type, size_t numSlots) : type_(std::move(type)) {
```

Comment:
I think this is okay to resolve the immediate issue, but I am pretty wary of a mixed ownership model like this. It's not possible to statically tell whether a given `Object` can be used on its own or not; concretely, there's nothing stopping a user from grabbing an `Object` constant from a frozen graph and trying to use it standalone, unaware that it is not possible to do so.

Given the relatively restricted use case for weak ownership, it's probably fine. But for my peace of mind, let's at least add some clear comments here about:
1. Why we have weak vs. strong objects.
2. What are the behavioral differences between weak vs. strong objects.
3. When you would use weak vs. strong objects.